code_of_conduct.md

```diff
@@ -0,0 +1,134 @@
+
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+We as members, contributors, and leaders pledge to make participation in our
+community a harassment-free experience for everyone, regardless of age, body
+size, visible or invisible disability, ethnicity, sex characteristics, gender
+identity and expression, level of experience, education, socio-economic status,
+nationality, personal appearance, race, caste, color, religion, or sexual identity
+and orientation.
+
+We pledge to act and interact in ways that contribute to an open, welcoming,
+diverse, inclusive, and healthy community.
+
+## Our Standards
+
+Examples of behavior that contributes to a positive environment for our
+community include:
+
+* Demonstrating empathy and kindness toward other people
+* Being respectful of differing opinions, viewpoints, and experiences
+* Giving and gracefully accepting constructive feedback
+* Accepting responsibility and apologizing to those affected by our mistakes,
+  and learning from the experience
+* Focusing on what is best not just for us as individuals, but for the
+  overall community
+
+Examples of unacceptable behavior include:
+
+* The use of sexualized language or imagery, and sexual attention or
+  advances of any kind
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email
+  address, without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Community leaders are responsible for clarifying and enforcing our standards of
+acceptable behavior and will take appropriate and fair corrective action in
+response to any behavior that they deem inappropriate, threatening, offensive,
+or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject
+comments, commits, code, wiki edits, issues, and other contributions that are
+not aligned to this Code of Conduct, and will communicate reasons for moderation
+decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior
+may be reported to the community leaders responsible for enforcement
+at [graphblas@lists.lbl.gov](graphblas@lists.lbl.gov).  All complaints
```

Comment:
Who are defined as the community leaders that would be responsible for “enforcement”?  Does it start with the steering committee?  I am not sure that the community-wide mailing list is the correct first step to reporting.

Reply:
Agreed, I put the mailing list address as a first approximation.  Perhaps the steering committee is the best choice as it already exists.

The ultimate punishment, after being escalated by several violations, is a total ban on community contribution.  Maybe this is as simple as removing from the mailing list and using github blocking:

https://docs.github.com/en/communities/maintaining-your-safety-on-github/blocking-a-user-from-your-organization

and that's it, I mean, there's only so much we can do.  

On that note, almost all other projects have CoCs already, for example redis (which also uses the Contributor Covenant):

https://redis.io/topics/governance

Anaconda:

https://www.anaconda.com/community-code-of-conduct

Given the cross language/database/ecosystem role that the GraphBLAS plays we are almost held to a higher standard because we must endeavor to respect them all.

-Michel